xbmc/cores/VideoPlayer/VideoRenderers/DebugRenderer.h

```diff
@@ -10,31 +10,39 @@
 
 #include "DebugInfo.h"
 #include "OverlayRenderer.h"
+#include "cores/VideoPlayer/DVDSubtitles/SubtitlesAdapter.h"
 
+#include <memory>
 #include <string>
 
-class CDVDOverlayText;
-
 class CDebugRenderer
 {
 public:
   CDebugRenderer();
   virtual ~CDebugRenderer();
+  void Initialize();
+  void Dispose();
   void SetInfo(DEBUG_INFO_PLAYER& info);
   void SetInfo(DEBUG_INFO_VIDEO& video, DEBUG_INFO_RENDER& render);
   void Render(CRect& src, CRect& dst, CRect& view);
   void Flush();
 
 protected:
-
   class CRenderer : public OVERLAY::CRenderer
   {
   public:
     CRenderer();
     void Render(int idx) override;
+    void CreateSubtitlesStyle();
+
+  private:
+    std::shared_ptr<struct KODI::SUBTITLES::subtitlesStyle> m_debugOverlayStyle;
   };
 
   std::string m_strDebug[6];
-  CDVDOverlayText* m_overlay[6];
   CRenderer m_overlayRenderer;
+
+private:
+  CSubtitlesAdapter* m_adapter{nullptr};
+  bool m_isInitialized{false};
```

Comment:
I am not sure if a simple bool is enough here as it will be consumed without any sort of locking by different threads. Maybe a `std::atomic_bool ` would make more sense?